src/stest/tests/set-up-character-special-file

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+# main : () -> string
+# Return a path to a character special file that exists on most systems.
+function main {
+	echo "/dev/tty" || return 1
```

Comment:
Same as above. It bothers me that we're making assumptions about existing files on the file system in this test and not just creating the test conditions ourselves.

Reply:
Neither `/dev/tty` nor `/dev/disk/by-diskseq/1` are actually files; they are guaranteed to exist by Linux.
> This statement is almost certainly wrong in some crazy edge case, but the assumption is reasonable. It's appropriate to go forward this way and change if someone files a bug down the road.

Reply:
> This statement is almost certainly wrong in some crazy edge case, but the assumption is reasonable. It's appropriate to go forward this way and change if someone files a bug down the road.

I agree. I'll plan to move forward with this until something better comes up.

One crazy edge-case that's on my mind, however, is the [sandboxed environment of a nix build][1]. But that can come later.

[1]: https://discourse.nixos.org/t/survey-how-are-you-using-flakes-today/15494/4

Reply:
Do nix builds commonly run tests as well? If not, this is a non-issue.

Reply:
I expect that many nix builds inside of the `nixpkgs` repository do end up skipping tests for one reason or another, however it requires an explicit disablement and I think it's probably discouraged.

For example, I disabled the test phase of `dmenu-rs`'s `make` build when checking it into `nixpkgs` [here][1]. There was some discussion in the PR from contributors about ways to emulate X to enable the tests to run, but ultimately it proved too much to hack around. But I hope that can change in the future.

[1]: https://github.com/NixOS/nixpkgs/blob/28319deb5ab05458d9cd5c7d99e1a24ec2e8fc4b/pkgs/applications/misc/dmenu-rs/default.nix#L79